flytekit/core/type_engine.py

```diff
@@ -731,6 +734,30 @@ def to_python_value(self, ctx: FlyteContext, lv: Literal, expected_python_type:
         return expected_python_type(lv.scalar.primitive.string_value)
 
 
+def convert_json_schema_to_python_class(schema):
+    """Generate a model class based on the provided JSON Schema
+    :param schema: dict representing valid JSON schema
+    """
+    schema = copy.deepcopy(schema)
+
+    class Model(dict):
+        def __init__(self, *args, **kwargs):
+            self.__dict__["schema"] = schema
+            d = dict(*args, **kwargs)
+            dict.__init__(self, d)
+
+        def __setitem__(self, key, value):
+            dict.__setitem__(self, key, value)
+
+        def __getattr__(self, key):
+            try:
+                return self.__getitem__(key)
```

Comment:
what if a nested item is a dataclass / schema too?

Reply:
It also works, check the new test below.